ugs-platform/ugs-platform-gcode-editor/src/main/java/com/willwinder/ugs/nbp/editor/parser/GcodeParserResult.java

```diff
@@ -0,0 +1,54 @@
+/*
+    Copyright 2016-2019 Will Winder
+
+    This file is part of Universal Gcode Sender (UGS).
+
+    UGS is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    UGS is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with UGS.  If not, see <http://www.gnu.org/licenses/>.
+*/
+package com.willwinder.ugs.nbp.editor.parser;
+
+import org.netbeans.modules.csl.spi.ParserResult;
+import org.netbeans.modules.parsing.api.Snapshot;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Gcode parser results holding all errors found
+ *
+ * @author Joacim Breiler
+ */
+public class GcodeParserResult extends ParserResult {
+
+    private final List<GcodeError> errorList;
+
+    public GcodeParserResult(Snapshot snapshot) {
+        super(snapshot);
+        errorList = new ArrayList<>();
```

Comment:
What are the snapshots? If you implemented syntax errors would they be generated as part of the `GcodeParserResult`?

Reply:
I actually don't use snapshot, it's used in the super class org.netbeans.modules.csl.spi.ParserResult